ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ConnectorCommandResponseHandler.java

```diff
@@ -0,0 +1,120 @@
+
+  private ConnectorCommandResponseHandler() {
+  }
+
+  static void handlerCreateConnectorResponse(
+      final JsonObject connectorInfoEntity,
+      final CompletableFuture<CreateConnectorResult> cf
+  ) {
+    try {
+      final JsonObject connectorInfoJson = connectorInfoEntity.getJsonObject("info");
+      cf.complete(new CreateConnectorResultImpl(
+          connectorInfoJson.getString("name"),
+          new ConnectorTypeImpl(connectorInfoJson.getString("type")),
+          connectorInfoJson.getJsonObject("config").mapTo(Map.class)
+      ));
+    } catch (Exception e) {
+      cf.completeExceptionally(new IllegalStateException(
+          "Unexpected server response format. Response: " + connectorInfoEntity
+      ));
+    }
+  }
+
+  static void handleDropConnectorResponse(
+      final JsonObject dropConnectorResponseEntity,
+      final CompletableFuture<Void> cf
+  ) {
+    if (dropConnectorResponseEntity.getString("@type").equals("drop_connector")) {
+      cf.complete(null);
+    } else {
+      cf.completeExceptionally(new IllegalStateException(
+          "Unexpected server response format. Response: " + dropConnectorResponseEntity
+      ));
+    }
+  }
+
+  static void handleListConnectorsResponse(
+      final JsonObject connectorsListEntity,
+      final CompletableFuture<List<ConnectorInfo>> cf
+  ) {
+    final Optional<List<ConnectorInfo>> connectors =
+        getListConnectorsResponse(connectorsListEntity);
+    if (connectors.isPresent()) {
+      cf.complete(connectors.get());
+    } else {
+      cf.completeExceptionally(new IllegalStateException(
+          "Unexpected server response format. Response: " + connectorsListEntity));
+    }
+  }
+
+  static void handleDescribeConnectorsResponse(
+      final JsonObject connectorDescriptionEntity,
+      final CompletableFuture<ConnectorDescription> cf
+  ) {
+    try {
+      final JsonObject status = connectorDescriptionEntity.getJsonObject("status");
+      cf.complete(new ConnectorDescriptionImpl(
+          connectorDescriptionEntity.getString("connectorClass"),
+          connectorDescriptionEntity.getJsonArray("sources").getList(),
+          connectorDescriptionEntity.getJsonArray("topics").getList(),
+          new ConnectorTypeImpl(status.getString("type")),
+          status.getJsonObject("connector").getString("state")
+      ));
+    } catch (Exception e) {
+      cf.completeExceptionally(new IllegalStateException(
+          "Unexpected server response format. Response: " + connectorDescriptionEntity));
+    }
+  }
+
+  /**
+   * Attempts to parse the provided response entity as a {@code List<ConnectorInfo>}.
+   *
+   * @param connectorsEntity response entity
+   * @return optional containing parsed result if successful, else empty
+   */
+  private static Optional<List<ConnectorInfo>> getListConnectorsResponse(
+      final JsonObject connectorsEntity) {
+    try {
+      final JsonArray connectors = connectorsEntity.getJsonArray("connectors");
```

Comment:
A bit late for this PR but we've been wanting to change the pattern of how the Java client response interfaces relate to the server response objects for a while now, in order to avoid custom JSON parsing like this. Here's the ticket with context: https://github.com/confluentinc/ksql/issues/6042

We should really do this before the next time we add additional interfaces, but it's outside the scope for this PR. (If you're keen on helping clean up the codebase and want to tackle it as a follow-up PR that'd be super, but no pressure 😁 )